akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -726,3 +729,139 @@ private[akka] object Log {
   private final val OffInt = LogLevels.Off.asInt
   private final val DefaultLogLevels = LogLevels(onElement = Logging.DebugLevel, onFinish = Logging.DebugLevel, onFailure = Logging.ErrorLevel)
 }
+
+/**
+ * INTERNAL API
+ */
+private[stream] object TimerKeys {
+  case object TakeWithinTimerKey
+  case object DropWithinTimerKey
+  case object GroupedWithinTimerKey
+}
+
+/**
+ * INTERNAL API
+ */
+private[stream] final case class GroupedWithin[T](n: Int, d: FiniteDuration) extends TimerStage[T, immutable.Seq[T]] {
+  import TimerKeys._
+  private var buf: Vector[T] = Vector.empty
+
+  override def preStart(ctx: AsyncContext[immutable.Seq[T], Scheduled]) = {
+    super.preStart(ctx)
+    schedulePeriodically(GroupedWithinTimerKey, d)
```

Comment:
Just a sanity check for myself:
- since `schedule` will be moved to `Materializer`, I can use it with AsyncStage - if my stage is rather simple I think I'd use that still.
- it's also usable with the GraphStage

win-win and everybody is happy, right?